packages/transaction-manager/README.md

```diff
@@ -1,24 +1,73 @@
-# `@ckb-lumos/transaction-manager`
-
-TransactionManager is a tool for manager uncommitted cells, you can `send_transaction` via this tool and get uncommitted outputs by `collector`.
+# `@ckb-lumos/pending-transactions-manager`
 
 ## Usage
 
-```javascript
-const TransactionManager = require("@ckb-lumos/transaction-manager");
-const { Indexer } = require("@ckb-lumos/ckb-indexer");
+You can create a `PendingTransactionsManager` by passing a `rpcUrl` and an optional `options` object.
+
+```ts
+  delcare const queryOptions: CkbIndexerQueryOptions;
+  const manager = new PendingTransactionsManager({
+    providers: {
+      rpcUrl: "https://testnet.ckb.dev",
+    }
+  });
+
+  const cellCollector = await manager.collector(queryOptions);
+  for await (const cell of cellCollector.collect()) {
+    // do something with the cell
+  }
+```
+
+> Note: `CkbIndexerQueryOptions` used in `transaction-manager` is a partial type of `CkbIndexerQueryOptions` used in `@ckb-lumos/indexer`. The field `skip` is supressed when generating a cell collector `collector(queryOptions)`, and when `usePendingCells` is set to `true`, the field `fromBlock` and `toBlock` are ignored, pending cells will be returned regardless of the block number. The order of the returned cells are on-chain cells come first, followed by pending cells when `order` is set to `asc`(default value is `asc`)， and vice versa.
 
-// generate a new `TransactionManager` instance and start.
-const indexer = new Indexer("http://127.0.0.1:8114");
-const transactionManager = new TransactionManager(indexer);
-transactionManager.start();
+By default the pending transactions are stored in memory, which means the pending tx infos will be lost if the user refreshes the browser.
 
-// now you send transaction via `transactionManager`.
-const txHash = await transactionManager.send_transaction(transaction);
+To persist the transactions, you can pass a `txStorage` option to the `PendingTransactionsManager` constructor.
 
-// you can get uncommitted cells by `transactionManager.collector`.
-const collector = transactionManager.collector({ lock });
-for await (const cell of collector.collect()) {
-  console.log(cell);
-}
+```ts
+  const manager = new PendingTransactionsManager({
+    providers: {
+      rpcUrl: "https://testnet.ckb.dev",
+    }
+    options: {
+      txStorage:  new PendingTransactionStorage(YOUR_STORAGE),
+    },
+  });
 ```
+
+Especially in browser enviroment, if you want to use `localStorage` as the storage, you can create a `PendingTransactionStorage` instance like this:
+
+
+```ts
+  const manager = new PendingTransactionsManager({
+    providers: {
+      rpcUrl: "https://testnet.ckb.dev",
+    }
+    options: {
+      txStorage:  new PendingTransactionStorage(createBrowserStorage()),
+    },
+  });
+
+  function createBrowserStorage() {
+    const store: Storage = window.localStorage;
+    return {
+      getItem(key) {
+        const value = store.getItem(key) as string | undefined;
```

Comment:
we should add tips for custom key to avoid conflict keys

## Avoid Storage Conflict

```ts
const CUSTOM_KEY_PREFIX = '__lumos_store__'

getItem(key) {
  const value = store.getItem(CUSTOM_KEY_PREFIX + key)
}
...
```

Reply:
updated